toolkit/tools/internal/pkggraph/pkggraph.go

```diff
@@ -1332,11 +1412,46 @@ func (g *PkgGraph) fixIntraSpecCycle(trimmedCycle []*PkgNode) (err error) {
 	return
 }
 
-// fixPrebuiltSRPMsCycle attempts to fix a cycle if at least one node is a pre-built SRPM.
-// If a cycle can be fixed, edges representing the build dependencies of the pre-built SRPM will be removed.
-func (g *PkgGraph) fixPrebuiltSRPMsCycle(trimmedCycle []*PkgNode) (err error) {
+/*runNode is the currentNode which is being replaced in the cycle, and buildNode is its previous Node in the cycle*/
+func (g *PkgGraph) replaceCurrentRunNodeWithNewNode(runNode, newNode, buildNode *PkgNode) {
```

Comment:
The function does what it should be doing, but the name is misleading. We're not simply replacing any one node with another. We're limiting the replacement in two ways:
1. We're only interested in nodes associated with one specific SRPM - the `parentNode.SrpmPath == buildNode.SrpmPath` check.
1. For that SRPM, we're only interested in replacing the dependencies from the **build** nodes - the `parentNode.Type == TypeLocalBuild` check.

With that in mind I suggest these changes:
1. Give the function a name closer representing its function. One idea: `replaceSRPMBuildDependency`.
2. Pass only the SRPM's path as the third argument instead of the whole node - the node itself is irrelevant, we only care about the SRPM.
3. Make the first argument's names more generic. For instance: `replacedNode`.